src/generator/operations.ts

```diff
@@ -423,19 +423,50 @@ function createProtocolRequest(client: string, op: Operation, imports: ImportMan
   const returns = ['*azcore.Request', 'error'];
   let text = `${comment(name, '// ')} creates the ${info.name} request.\n`;
   text += `func (client *${client}) ${name}(${getCreateRequestParametersSig(op)}) (${returns.join(', ')}) {\n`;
+  let includeParse = false;
+  let parseVar = '';
+  // hostURLStr will be used to check for the x-ms-parameterized-host functionality
+  const hostURLStr = <string>op.requests![0].protocol.http!.uri;
+  // for parameterized hosts that only in the format {x} will be ignored and the client url will be used
+  if (!((hostURLStr.match(/{/g) || []).length === 1 && (hostURLStr.match(/}/g) || []).length === 1 && hostURLStr.indexOf('{') === 0 && hostURLStr.indexOf('}') === hostURLStr.length - 1)) {
+    text += `\thost := "${hostURLStr}"\n`;
+    // replace url parameters
+    for (const pp of values(aggregateParameters(op)).where((each: Parameter) => { return each.protocol.http !== undefined && each.protocol.http!.in === 'uri'; })) {
+      imports.add('strings');
+      // host references may be found among other variables and therefore call to use the host 
+      // specified by the endpoint the client was created with
+      if (pp.language.go!.name == 'host') {
```

Comment:
Can't this be removed?  I think it's dangerous to rely on a specific name (also should be `===`).

Reply:
I'll double check the swaggers, to see if this case still exists.

Reply:
So there is a test in the testserver that relies on this. Actually it's the same as the one below: https://github.com/Azure/autorest.testserver/blob/master/swagger/custom-baseUrl.json

Without other identifying information, I'm not sure how to go about determining what params are host values. I agree it's not good to check based off of the name. :(

Reply:
Thinking about this, given this is parameterizing the host I think it's safe to say that the first path param in `hostURLStr` will always be the host and should be replaced with `client.u.Host` so we don't need to rely on the param name.  Any other params will use named replacement.  What do you think?

Reply:
Agreed this was removed.